### ISSUE TITLE
AnnotationsSortedAlphabetically: New Check to have all annotations sorted alphabetically

### ISSUE THREAD

**Issue by ilopmar** [checkstyle/checkstyle#11693]
In Java it's considered a good practice to sort all imports alphabetically. IDEs have been doing that for a long time. 

In our team we have some Coding Standards and one of them is also keeping the annotations in classes/fields/methods,... sorted alphabetically. There is no way we can do that automatically so in every merge request we need to manually check it and, as humans, sometimes we missed some of them.
We would like to have a CheckStyle rule to enforce that the annotations are sorted alphabetically so the build can fail if not.

The name of the new Check could be *AnnotationsSortedAlphabetically*.

Given the following configuration:

```xml
<module name="AnnotationsSortedAlphabetically" />
```

With a class like:

```java
@SuppressWarnings("foo")  // invalid
@Deprecated
class Foo {

  @Deprecated  // valid
  @SuppressWarnings("bar")
  public static void main(String[] args) {
    
  }
}
```

I envision the output like:

```
...
[INFO] --- maven-checkstyle-plugin:3.1.2:check (default-cli) @ accounting-service-core ---
[INFO] There is 1 error reported by Checkstyle 10.1 with ../src/main/config/checkstyle-java-google-style.xml ruleset.
[ERROR] src/main/java/..../Foo.java:[1,1] (extension) AnnotationsSortedAlphabetically: Annotations "SuppressWarnings, Deprecated" not sorted alphabetically
...
```

I have a working Check that I created using current annotation checks as a "template". The code is a little bit ugly and I need to figure out how to properly test it, but, if you like this idea, I can contribute what I've done, and with your help, make the code better so it finally get accepted.

**Comment by Vyom-Yadav** [comment]
@romani @rnveach @strkkk @nick-mancuso @pbludov Seems a reasonable request to me. Also, I found this has decent support on stack overflow as well -> https://stackoverflow.com/q/32548869/15412365

**Comment by strkkk** [comment]
As for me it seems that such check can be more generic. I mean that it can be something like `CustomAnnotationsOrderCheck` with some predefined set of order strategies - alphabet order, by length (to force annotations with long name or parameter list to be at the end) or force some annotation to be first/last in list.

**Comment by romani** [comment]
@ilopmar , thanks a for sharing idea.

>In Java it's considered a good practice to sort all imports alphabetically. 

to be precise: do grouping and sort inside groups alphabetically.
User will need the same. To group Lombok, jdk, jackson annotatons. and do not mix them.
But we can start with simple ordering for now, if community get more interest we cacn create more addvanced Check

if you contribute to this repo, you will face very detailed review and high level of quality.
Alternatively we have https://github.com/sevntu-checkstyle/sevntu.checkstyle repo, where we more relaxed and ok to experiment.
